app/code/core/Mage/Core/Model/Resource/Abstract.php

```diff
@@ -245,4 +279,31 @@ protected function _prepareTableValueForSave($value, $type)
         }
         return $value;
     }
+
+    /**
+     * Prepare transaction isolation level for session.
+     *
+     * @param int $isolationLevel Any of the supported isolation levels defined in this class.
+     * @return void
+     */
+    public function prepareTransactionIsolationLevel($isolationLevel)
+    {
+        if( isset(self::SUPPORTED_ISOLATION_LEVELS[$isolationLevel]) && self::$_currentIsolationLevel != $isolationLevel) {
```

Comment:
Add a new method `isIsolationLevelSupported(int $isolationLevel): bool` then throw an exception here if it is not supported.

"Prepare isolation level" makes me think it doesn't actually do anything. This is actually _setting_ the isolation level. Change the name to that?